app/channel.go

```diff
@@ -1600,7 +1600,14 @@ func (a *App) MarkChannelsAsViewed(channelIds []string, userId string, clearPush
 	channelsToClearPushNotifications := []string{}
 	if *a.Config().EmailSettings.SendPushNotifications && clearPushNotifications {
 		for _, channelId := range channelIds {
-			result := <-a.Srv.Store.Channel().GetMember(channelId, userId)
+			result := <-a.Srv.Store.Channel().Get(channelId, true)
+			if result.Err != nil {
+				mlog.Warn(fmt.Sprintf("Failed to get channel %v", result.Err))
+				continue
+			}
+			channel := result.Data.(*model.Channel)
+
+			result = <-a.Srv.Store.Channel().GetMember(channelId, userId)
```

Comment:
I think we wanted no shadowing except for `err`.

Reply:
@crspeller changes made